### ISSUE TITLE
Attempt to invoke virtual method 'void java.net.DatagramSocket.setReuseAddress(boolean)' on a null object reference com.tradle.react.UdpSocketClient.bind

### ISSUE THREAD

**Issue by rimzici** [tradle/react-native-udp#119]
Seeing numerous crashes in the release build.

**VERSION**

 "react": "16.9.0",
 "react-native": "0.61.5",
"react-native-udp": "^3.0.0",

<details>
 <summary>Full crash log</summary>

```java
Fatal Exception: java.lang.RuntimeException: An error occurred while executing doInBackground()
       at android.os.AsyncTask$3.done(AsyncTask.java:353)
       at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:383)
       at java.util.concurrent.FutureTask.setException(FutureTask.java:252)
       at java.util.concurrent.FutureTask.run(FutureTask.java:271)
       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162)
       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636)
       at java.lang.Thread.run(Thread.java:764)
Caused by java.lang.NullPointerException: Attempt to invoke virtual method 'void java.net.DatagramSocket.setReuseAddress(boolean)' on a null object reference
       at com.tradle.react.UdpSocketClient.bind(UdpSocketClient.java:83)
       at com.tradle.react.UdpSockets$3.doInBackgroundGuarded(UdpSockets.java:140)
       at com.tradle.react.UdpSockets$3.doInBackgroundGuarded(UdpSockets.java:131)
       at com.facebook.react.bridge.GuardedAsyncTask.doInBackground(GuardedAsyncTask.java:35)
       at com.facebook.react.bridge.GuardedAsyncTask.doInBackground(GuardedAsyncTask.java:19)
       at android.os.AsyncTask$2.call(AsyncTask.java:333)
       at java.util.concurrent.FutureTask.run(FutureTask.java:266)
       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162)
       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636)
       at java.lang.Thread.run(Thread.java:764)

```
</details>

**OS**
Reported in Android 7, 8, 9 and 10.

**SAMPLE USAGE**

Sample of how the plugin is used:

```js

let socket = {};
// Establish new UDP socket only after closing the existing socket completely.
closeUDPSocket(() => {
    socket = dgram.createSocket({
        type: 'udp4',
	reuseAddr: true,
	reusePort: true,
	});

	const aPort = randomPort();

	socket.on('error', (error: any) => {
	});

	try {
		socket.bind(aPort);
	} catch (error) {
	}

	socket.once('listening', () => {
         });
});

function closeUDPSocket(callback?: Function | null = null) {
	if (socket && socket.close) {
		socket.close(() => {
			socket = {};

			if (callback && typeof callback === 'function') {
				callback();
			}
		});
	} else if (callback && typeof callback === 'function') {
		socket = {};
		callback();
	}
}
```

**NOTE: `close` is called on any existing connection before new one is opened.**

As understandable from the log, it is [this line](https://github.com/tradle/react-native-udp/blob/master/android/src/main/java/com/tradle/react/UdpSocketClient.java#L83) causing the crash.

Would appreciate any help! Thanks!!

**Comment by baharb** [comment]
@rimzici Hi,
I have this problem too
Did you find any solution?

**Comment by rimzici** [comment]
If you would like, you can use [this fork](https://github.com/telldus/react-native-udp) for now.

**Comment by baharb** [comment]
> If you would like, you can use [this fork](https://github.com/telldus/react-native-udp) for now.

tnx @rimzici 
I'm using RN 0.57
Is this fork compatible with 0.57?

**Comment by Fr33maan** [comment]
@rimzici does the change you made in the fork fix the issue ? If so would you consider open a PR ?

**Comment by Rapsssito** [comment]
@rimzici, could you try if you experience the same issue using the master branch? I have merged a PR (#108) with a possible fix, but it has not been published yet.

**Comment by hubciorz** [comment]
@Rapsssito This fix does not work. The problem is still occurring on 4.0.3

**Comment by Fr33maan** [comment]
@hubciorz can you give more information about your env

**Comment by hubciorz** [comment]
@Fr33maan Problem still occurs on 4.0.4. Try/catch fix by @rimzici works.

**Comment by Rapsssito** [comment]
@hubciorz, could you create a new issue with the error trace, the code causing the error and the reference to the solution proposed by @rimzici?
